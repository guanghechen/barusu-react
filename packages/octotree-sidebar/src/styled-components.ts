import type { OctotreeSidebarTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    octotreeSidebar?: OctotreeSidebarTheme
  }
}
