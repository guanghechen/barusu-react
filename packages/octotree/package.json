{
  "name": "@barusu-react/octotree",
  "version": "0.0.19",
  "description": "render a directory tree like the chrome extension 'octotree'",
  "author": {
    "name": "guanghechen",
    "url": "https://github.com/guanghechen/"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/guanghechen/barusu-react.git"
  },
  "homepage": "https://github.com/guanghechen/barusu-react/tree/master/packages/octotree#readme",
  "keywords": [],
  "main": "lib/cjs/index.js",
  "module": "lib/esm/index.js",
  "types": "lib/types/index.d.ts",
  "source": "src/index.tsx",
  "license": "MIT",
  "engines": {
    "node": ">= 14.15.0"
  },
  "files": [
    "lib/",
    "CHANGELOG.md",
    "LICENSE",
    "README.md",
    "package.json",
    "tsconfig.json",
    "!lib/**/*.js.map",
    "!lib/**/*.d.ts.map"
  ],
  "scripts": {
    "start": "rollup -w -c rollup.config.js",
    "prebuild": "rimraf lib/",
    "build": "cross-env NODE_ENV=production rollup -c rollup.config.js",
    "prepublishOnly": "cross-env ROLLUP_USE_SOURCE_MAP=false yarn build",
    "test": "cross-env TS_NODE_FILES=true jest --config ../../jest.config.js --rootDir .",
    "lint:fix": "eslint '{src,__test__}/**/*.{ts,tsx}' --fix"
  },
  "peerDependencies": {
    "@types/react": "^16.9.56",
    "@types/react-router-dom": "^5.1.6",
    "@types/styled-components": "^5.1.4",
    "prop-types": "^15.7.2",
    "react": "^16.14.0",
    "react-router-dom": "^5.2.0",
    "styled-components": "^5.2.1"
  },
  "devDependencies": {
    "@types/react": "^16.9.56",
    "@types/react-router-dom": "^5.1.6",
    "@types/styled-components": "^5.1.4",
    "prop-types": "^15.7.2",
    "react": "^16.14.0",
    "react-router-dom": "^5.2.0",
    "rollup": "2.35.1",
    "styled-components": "^5.2.1"
  },
  "browserslist": [
    "last 2 versions",
    "Firefox ESR",
    "> 1%",
    "ie >= 11"
  ]
}
