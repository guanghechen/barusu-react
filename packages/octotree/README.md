<header>
  <h1 align="center">
    <a href="https://github.com/guanghechen/barusu-react/tree/master/packages/octotree#readme">@barusu-react/octotree</a>
  </h1>
  <div align="center">
    <a href="https://www.npmjs.com/package/@barusu-react/octotree">
      <img
        alt="Npm Version"
        src="https://img.shields.io/npm/v/@barusu-react/octotree.svg"
      />
    </a>
    <a href="https://www.npmjs.com/package/@barusu-react/octotree">
      <img
        alt="Npm Download"
        src="https://img.shields.io/npm/dm/@barusu-react/octotree.svg"
      />
    </a>
    <a href="https://www.npmjs.com/package/@barusu-react/octotree">
      <img
        alt="Npm License"
        src="https://img.shields.io/npm/l/@barusu-react/octotree.svg"
      />
    </a>
    <a href="https://github.com/nodejs/node">
      <img
        alt="Node.js Version"
        src="https://img.shields.io/node/v/@barusu-react/octotree"
      />
    </a>
    <a href="https://github.com/facebook/react">
      <img
        alt="React version"
        src="https://img.shields.io/npm/dependency-version/@barusu-react/octotree/peer/react"
      />
    </a>
    <a href="https://github.com/facebook/jest">
      <img
        alt="Tested with Jest"
        src="https://img.shields.io/badge/tested_with-jest-9c465e.svg"
      />
    </a>
    <a href="https://github.com/prettier/prettier">
      <img
        alt="Code Style: prettier"
        src="https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square"
      />
    </a>
  </div>
</header>
<br/>


Render a simple directory tree similar to [octotree][], but really more coarse.

See [sourcecodes in github][homepage] to for details.

## Install

* npm

  ```bash
  npm install --save @barusu-react/octotree
  ```

* yarn

  ```bash
  yarn add @barusu-react/octotree

## Usage

  You should import the [octotree.woff2][] font manually.

  ```css
  @font-face {
    src: url('/font/octotree.woff2') format('woff2');
    font-family: 'octotree';
  }
  ```

  * Use in React project

    - Pure

      ```tsx
      import React from 'react'
      import { BrowserRouter as Router } from 'react-router-dom'
      import Octotree, { resolveOctotreeData } from '@barusu-react/octotree'

      const data = resolveOctotreeData([
        {
          title: 'hooks',
          children: [
            {
              title: 'demo',
              pathname: '/hooks/demo',
              children: [{ title: '2', pathname: '/hooks/demo/2' }]
            },
            { title: 'ref', pathname: '/hooks/ref' }
          ]
        },
        { title: 'context', pathname: '/context' },
        { title: 'demo', pathname: '/demo' },
      ])

      const wrapper = (
        <Router>
          <Octotree
            nodes={ data }
            iconWidth="1.5rem"
            style={{ color: 'orange', fontSize: '16px' }}
          />
        </Router>
      )
      ```

    - With theme

      ```tsx
      import React from 'react'
      import { BrowserRouter as Router } from 'react-router-dom'
      import { DefaultTheme, ThemeProvider } from 'styled-components'
      import Octotree, { resolveOctotreeData } from '@barusu-react/octotree'

      const data = resolveOctotreeData([
        {
          title: 'hooks',
          children: [
            {
              title: 'demo',
              pathname: '/hooks/demo',
              children: [{ title: '2', pathname: '/hooks/demo/2' }]
            },
            { title: 'ref', pathname: '/hooks/ref' }
          ]
        },
        { title: 'context', pathname: '/context' },
        { title: 'demo', pathname: '/demo' },
      ])

      const theme: DefaultTheme = {
        octotree: {
          fontSize: '18px',
          colorTitle: '#ccc',
          linkBackgroundHover: 'red',
          linkBackgroundActive: 'blue',
          typeIconColorSecondary: 'green'
        }
      }}

      const wrapper = (
        <ThemeProvider theme={ theme }>
          <Router>
            <Octotree
              nodes={ data }
              iconWidth="1.5rem"
              style={{ color: 'orange', fontSize: '16px' }}
            />
          </Router>
        </ThemeProvider>
      )
      ```

  * Props

     Name         | Type                              | Required  | Default   | Description
    :------------:|:---------------------------------:|:---------:|:---------:|:-------------
     `ref`        | `React.RefObject<HTMLDivElement>` | `false`   | -         | Forwarded ref callback
     `nodes`      | `OctotreeNodeData[]`              | `true`    | -         | Node data of octotree
     `iconWidth`  | `string`                          | `false`   | `1.25rem` | Icon width

    OctotreeProps inherited all attributes of `HTMLDivElement` (`React.HTMLAttributes<HTMLDivElement>`)

  * Theme

     Prop Name              | Default
    :----------------------:|:--------------
     fontSize               | `1rem`
     colorTitle             | `#0f2e47`
     linkBackgroundHover    | `#f6f8fa`
     linkBackgroundActive   | `#dbeeff`
     typeIconColorPrimary   | `#6cb5fe`
     typeIconColorSecondary | `#8a8a8a`

    See [OctotreeTheme][] for details.


[homepage]: https://github.com/guanghechen/barusu-react/tree/master/packages/octotree#readme
[octotree]: https://github.com/ovity/octotree.git
[OctotreeTheme]: https://github.com/guanghechen/barusu-react/blob/master/packages/octotree/src/theme.ts
[octotree.woff2]: https://github.com/ovity/octotree/blob/c8819379c9cc60b3c2124440766906028891120d/libs/fonts/octicons.woff2
