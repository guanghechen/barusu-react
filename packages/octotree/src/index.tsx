import './styled-components'
export * from './_common'
export * from './leaf'
export * from './parent'
export * from './theme'
export * from './tree'
export { default } from './tree'
export * from './types'
export * from './util'
