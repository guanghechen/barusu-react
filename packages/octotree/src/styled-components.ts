import type { OctotreeTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    octotree?: OctotreeTheme
  }
}
