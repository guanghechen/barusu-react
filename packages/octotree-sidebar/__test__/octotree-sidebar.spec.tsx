import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import { RawOctotreeNodeData, resolveOctotreeData } from '@barusu-react/octotree'
import OctotreeSidebar from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  const rawData: RawOctotreeNodeData[] = [
    {
      title: 'hooks',
      children: [
        {
          title: 'demo',
          pathname: '/hooks/demo',
          children: [{ title: '2', pathname: '/hooks/demo/2' }]
        },
        { title: 'ref', pathname: '/hooks/ref' }
      ]
    },
    { title: 'context', pathname: '/context' },
    { title: 'demo', pathname: '/demo' },
  ]

  const data = resolveOctotreeData('/api', rawData)

  it('nodes is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => render(
        <Router>
          <OctotreeSidebar nodes={ value } />
        </Router>
      ))
        .toThrow(/Failed prop type: The prop `nodes` is marked as required/i)
    }
  })

  it('hover', () => {
    const wrapper = mount(
      <Router>
        <OctotreeSidebar nodes={ data } initialWidth={ 120 } />
      </Router>
    )
    const o = wrapper.getDOMNode()
    expect(getComputedStyle(o).width).toBe('0px')

    wrapper.simulate('mouseenter')
    // expect(getComputedStyle(o).width).toBe('120px')
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLDivElement>()
    const wrapper = mount(
      <Router>
        <OctotreeSidebar ref={ ref } data-value="waw" nodes={ data } />
      </Router>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <Router>
        <OctotreeSidebar
          nodes={ data }
          initialWidth={ 250 }
          style={{ color: 'orange', fontSize: '16px' }}
        />
      </Router>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      octotreeSidebar: {
        borderRight: 'none',
        headerBackground: 'red',
        // mainBackground: '#fff',
        toggleBackground: 'blue',
        toggleBorderColor: 'green',
      }
    }

    const wrapper = render(
      <ThemeProvider theme={ theme }>
        <Router>
          <OctotreeSidebar
            nodes={ data }
            initialWidth={ 250 }
            style={{ color: 'orange', fontSize: '16px' }}
          />
        </Router>
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
