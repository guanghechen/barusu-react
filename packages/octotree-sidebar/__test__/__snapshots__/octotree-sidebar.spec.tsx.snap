// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c10 {
  display: inline-block;
  font-size: 1rem;
  font-family: octotree;
  font-style: normal;
  text-align: center;
}

.c10::before {
  font-size: inherit;
  font-style: normal;
}

.c13 {
  overflow: hidden;
  text-overflow: ellipsis;
  color: #0f2e47;
  font-weight: normal;
  font-style: normal;
}

.c17 {
  color: #8a8a8a;
}

.c17::before {
  content: "\\e827";
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  font-size: inherit;
  white-space: nowrap;
  background: none;
}

.c16:hover {
  background: #f6f8fa;
}

.c11::before {
  content: '\\e808';
}

.c12 {
  color: #6cb5fe;
}

.c12::before {
  content: '\\e840';
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c8:hover {
  background: #f6f8fa;
}

.c15 {
  display: block;
  overflow: auto;
  height: auto;
  opacity: 1;
}

.c7 .c9::before {
  content: '\\e80a';
}

.c7 .c14 {
  height: 0;
  opacity: 0;
}

.c6 {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
}

.c6 ul,
.c6 li {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
  line-height: 1.5;
  font-size: 1rem;
}

.c6 a {
  color: #0f2e47;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
  text-indent: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c18 {
  position: absolute;
  background: transparent;
  z-index: 99;
}

.c2 {
  position: absolute;
  top: 50%;
  right: 1rem;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  color: #fff;
}

.c2::before {
  content: "\\e829";
  display: inline-block;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  width: 1rem;
  line-height: 1;
  font-size: 1rem;
  font-family: octotree;
  font-style: normal;
  font-weight: 400;
  text-align: center;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  font-variant: normal;
}

.c1 {
  -webkit-flex: 0 0 2rem;
  -ms-flex: 0 0 2rem;
  flex: 0 0 2rem;
  position: relative;
  background: #373e43;
}

.c4 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  overflow: auto;
  background: #fff;
}

.c3 {
  position: absolute;
  top: 50%;
  left: 0;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  height: 6rem;
  width: 2rem;
  background: #f2f5f7;
  border: 1px solid #e0e4e7;
  border-left: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: default;
  display: none;
  opacity: 0;
}

.c3::before {
  font-family: octotree;
  content: "\\e80a";
}

.c3 > span {
  display: inline-block;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  writing-mode: tb-rl;
  color: #0f2e47;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  border: 0;
  border-right: 1px solid #dfe2e5;
  margin: 0;
  -webkit-transition: max-width 0.5s ease-in-out;
  transition: max-width 0.5s ease-in-out;
}

<div
  class="c0"
  style="color:orange;font-size:16px;width:250px;max-width:250px"
>
  <div
    class="c1"
  >
    <i
      class="c2"
    />
  </div>
  <div
    class="c3"
  >
    <span>
      Octotree
    </span>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <ul
        class="c6"
      >
        <li>
          <div
            class="c7"
          >
            <div
              class="c8"
              style="padding-left:0"
            >
              <i
                class="c9 c10 c11"
                style="width:1.25rem"
              />
              <i
                class="c10 c12"
                style="width:1.25rem;padding-right:0.25rem"
              />
              <h4
                class="c13"
                title="hooks"
              >
                hooks
              </h4>
            </div>
            <div
              class="c14 c15"
            >
              <ul>
                <li>
                  <div
                    class="c7"
                  >
                    <div
                      class="c8"
                      style="padding-left:1.25rem"
                    >
                      <i
                        class="c9 c10 c11"
                        style="width:1.25rem"
                      />
                      <i
                        class="c10 c12"
                        style="width:1.25rem;padding-right:0.25rem"
                      />
                      <h4
                        class="c13"
                        title="demo"
                      >
                        demo
                      </h4>
                    </div>
                    <div
                      class="c14 c15"
                    >
                      <ul>
                        <li>
                          <a
                            class="c16"
                            href="/api/hooks/demo/2"
                            style="padding-left:3.75rem"
                          >
                            <i
                              class="c10 c17"
                              style="width:1.25rem;padding-right:0.25rem"
                            />
                            <h4
                              class="c13"
                              title="2"
                            >
                              2
                            </h4>
                          </a>
                        </li>
                      </ul>
                    </div>
                  </div>
                </li>
                <li>
                  <a
                    class="c16"
                    href="/api/hooks/ref"
                    style="padding-left:2.50rem"
                  >
                    <i
                      class="c10 c17"
                      style="width:1.25rem;padding-right:0.25rem"
                    />
                    <h4
                      class="c13"
                      title="ref"
                    >
                      ref
                    </h4>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li>
          <a
            class="c16"
            href="/api/context"
            style="padding-left:1.25rem"
          >
            <i
              class="c10 c17"
              style="width:1.25rem;padding-right:0.25rem"
            />
            <h4
              class="c13"
              title="context"
            >
              context
            </h4>
          </a>
        </li>
        <li>
          <a
            class="c16"
            href="/api/demo"
            style="padding-left:1.25rem"
          >
            <i
              class="c10 c17"
              style="width:1.25rem;padding-right:0.25rem"
            />
            <h4
              class="c13"
              title="demo"
            >
              demo
            </h4>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="c18"
    style="top:0;bottom:0;left:unset;right:-4px;height:100%;width:8px;cursor:col-resize"
  />
</div>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c10 {
  display: inline-block;
  font-size: 1rem;
  font-family: octotree;
  font-style: normal;
  text-align: center;
}

.c10::before {
  font-size: inherit;
  font-style: normal;
}

.c13 {
  overflow: hidden;
  text-overflow: ellipsis;
  color: #0f2e47;
  font-weight: normal;
  font-style: normal;
}

.c17 {
  color: #8a8a8a;
}

.c17::before {
  content: "\\e827";
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  font-size: inherit;
  white-space: nowrap;
  background: none;
}

.c16:hover {
  background: #f6f8fa;
}

.c11::before {
  content: '\\e808';
}

.c12 {
  color: #6cb5fe;
}

.c12::before {
  content: '\\e840';
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c8:hover {
  background: #f6f8fa;
}

.c15 {
  display: block;
  overflow: auto;
  height: auto;
  opacity: 1;
}

.c7 .c9::before {
  content: '\\e80a';
}

.c7 .c14 {
  height: 0;
  opacity: 0;
}

.c6 {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
}

.c6 ul,
.c6 li {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
  line-height: 1.5;
  font-size: 1rem;
}

.c6 a {
  color: #0f2e47;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
  text-indent: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c18 {
  position: absolute;
  background: transparent;
  z-index: 99;
}

.c2 {
  position: absolute;
  top: 50%;
  right: 1rem;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  color: #fff;
}

.c2::before {
  content: "\\e829";
  display: inline-block;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  width: 1rem;
  line-height: 1;
  font-size: 1rem;
  font-family: octotree;
  font-style: normal;
  font-weight: 400;
  text-align: center;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  font-variant: normal;
}

.c1 {
  -webkit-flex: 0 0 2rem;
  -ms-flex: 0 0 2rem;
  flex: 0 0 2rem;
  position: relative;
  background: red;
}

.c4 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  overflow: auto;
  background: #fff;
}

.c3 {
  position: absolute;
  top: 50%;
  left: 0;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  height: 6rem;
  width: 2rem;
  background: blue;
  border: 1px solid green;
  border-left: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: default;
  display: none;
  opacity: 0;
}

.c3::before {
  font-family: octotree;
  content: "\\e80a";
}

.c3 > span {
  display: inline-block;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  writing-mode: tb-rl;
  color: #0f2e47;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  border: 0;
  border-right: none;
  margin: 0;
  -webkit-transition: max-width 0.5s ease-in-out;
  transition: max-width 0.5s ease-in-out;
}

<div
  class="c0"
  style="color:orange;font-size:16px;width:250px;max-width:250px"
>
  <div
    class="c1"
  >
    <i
      class="c2"
    />
  </div>
  <div
    class="c3"
  >
    <span>
      Octotree
    </span>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <ul
        class="c6"
      >
        <li>
          <div
            class="c7"
          >
            <div
              class="c8"
              style="padding-left:0"
            >
              <i
                class="c9 c10 c11"
                style="width:1.25rem"
              />
              <i
                class="c10 c12"
                style="width:1.25rem;padding-right:0.25rem"
              />
              <h4
                class="c13"
                title="hooks"
              >
                hooks
              </h4>
            </div>
            <div
              class="c14 c15"
            >
              <ul>
                <li>
                  <div
                    class="c7"
                  >
                    <div
                      class="c8"
                      style="padding-left:1.25rem"
                    >
                      <i
                        class="c9 c10 c11"
                        style="width:1.25rem"
                      />
                      <i
                        class="c10 c12"
                        style="width:1.25rem;padding-right:0.25rem"
                      />
                      <h4
                        class="c13"
                        title="demo"
                      >
                        demo
                      </h4>
                    </div>
                    <div
                      class="c14 c15"
                    >
                      <ul>
                        <li>
                          <a
                            class="c16"
                            href="/api/hooks/demo/2"
                            style="padding-left:3.75rem"
                          >
                            <i
                              class="c10 c17"
                              style="width:1.25rem;padding-right:0.25rem"
                            />
                            <h4
                              class="c13"
                              title="2"
                            >
                              2
                            </h4>
                          </a>
                        </li>
                      </ul>
                    </div>
                  </div>
                </li>
                <li>
                  <a
                    class="c16"
                    href="/api/hooks/ref"
                    style="padding-left:2.50rem"
                  >
                    <i
                      class="c10 c17"
                      style="width:1.25rem;padding-right:0.25rem"
                    />
                    <h4
                      class="c13"
                      title="ref"
                    >
                      ref
                    </h4>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li>
          <a
            class="c16"
            href="/api/context"
            style="padding-left:1.25rem"
          >
            <i
              class="c10 c17"
              style="width:1.25rem;padding-right:0.25rem"
            />
            <h4
              class="c13"
              title="context"
            >
              context
            </h4>
          </a>
        </li>
        <li>
          <a
            class="c16"
            href="/api/demo"
            style="padding-left:1.25rem"
          >
            <i
              class="c10 c17"
              style="width:1.25rem;padding-right:0.25rem"
            />
            <h4
              class="c13"
              title="demo"
            >
              demo
            </h4>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="c18"
    style="top:0;bottom:0;left:unset;right:-4px;height:100%;width:8px;cursor:col-resize"
  />
</div>
`;
